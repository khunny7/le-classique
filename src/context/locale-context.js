import React, { useState } from 'react';
import PropTypes from 'prop-types';
import GetLocaleTextLoader from '../text/text-loader';

const LocaleContext = React.createContext({
  currentLocale: 'en',
});

const LocaleContextProvider = (props) => {
  const { children } = props;
  const [currentLocale, setCurrentLocale] = useState('en');

  return (
    <LocaleContext.Provider value={
      {
        textLoader: GetLocaleTextLoader(currentLocale),
        setCurrentLocale: (locale) => setCurrentLocale(locale),
        currentLocale,
      }
    }
    >
      <div lang={currentLocale}>
        {children}
      </div>
    </LocaleContext.Provider>
  );
};

const withLocaleContext = (Component) => (
  (props) => (
    <LocaleContext.Consumer>
      {(context) => <Component localeContext={context} {...props} />}
    </LocaleContext.Consumer>
  )
);


LocaleContextProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

export { LocaleContext, LocaleContextProvider, withLocaleContext };
