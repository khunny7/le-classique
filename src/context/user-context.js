import React from 'react';
import PropTypes from 'prop-types';
import firebase from 'firebase/app';
import UserRepository from '../data/user-repository';

const UserContext = React.createContext({
  currentUser: {
    name: 'guest',
  },
});

class UserContextProvider extends React.Component {
  constructor(props) {
    super(props);

    this.auth = firebase.auth();

    this.auth.onAuthStateChanged((user) => {
      if (user) {
        UserRepository.getUserAndSet(user.uid, user).then((lcUser) => {
          this.setState({
            currentUser: lcUser.getData(),
          });
        });
      } else {
        this.setState({ currentUser: null });
      }
    });

    if (this.auth.currentUser) {
      UserRepository.getUserAndSet(this.auth.currentUser.uid, this.auth.currentUser).then((lcUser) => {
        this.setState({
          currentUser: lcUser.getData(),
        });
      });
    } else {
      this.state = {
        currentUser: null,
      };
    }
  }

  render() {
    const { children } = this.props;
    const { currentUser } = this.state;

    return (
      <UserContext.Provider value={{ currentUser, signOut: () => firebase.auth().signOut() }}>
        {children}
      </UserContext.Provider>
    );
  }
}

UserContextProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

const withUserContext = (Component) => (
  (props) => (
    <UserContext.Consumer>
      {(context) => <Component userContext={context} {...props} />}
    </UserContext.Consumer>
  )
);

export { UserContext, UserContextProvider, withUserContext };
