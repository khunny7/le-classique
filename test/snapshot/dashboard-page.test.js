import React from 'react';
import renderer from 'react-test-renderer';
import DashboardPage from '../../src/dashboard/dashboard-page';
import * as LocaleContext from '../../src/context/locale-context';

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

jest.mock('../../src/context/locale-context', () => ({
  useLocaleContext: () => ({
    textLoader: (key) => key,
    currentLocale: 'en',
  }),
}));

jest.mock('../../src/components/page-header', () => 'MockedPageHeader');

describe('<DashboardPage /> - snapshots', () => {
  test('DashboardPage ', () => {
    const component = renderer.create(
      <DashboardPage />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
