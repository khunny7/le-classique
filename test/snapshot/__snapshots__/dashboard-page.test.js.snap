// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardPage /> - snapshots DashboardPage  1`] = `
<div
  className="dashboard-container"
>
  <MockedPageHeader />
  <div
    className="jumbotron jumbotron"
  >
    <div
      className="welcome-user"
    >
      <div
        className="welcome-user-text"
      >
        <h1
          className="user-name"
          lang="en"
        >
          Hello_Users_Label
        </h1>
        <p
          className="app-description"
          lang="en"
        >
          App_Description_Label
        </p>
        <button
          className="browsing-btn btn btn-primary"
          disabled={false}
          type="button"
        >
          Book_Bookshelf_Button_Label
           ⯆
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <p
      lang="en"
    >
      Books_You_Read
    </p>
    <div
      className="book-list-view"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12"
        lang="en"
      >
        Books_All_Library
      </div>
    </div>
    <div
      className="book-list-view"
    />
  </div>
</div>
`;
